### DISCUSSION TITLE
Issues with Layout and Responsiveness Due to Overuse of Absolute Positioning

### DISCUSSION THREAD

**Discussion by ghost** [community/community#144010] [Programming Help]
### Body

I ran into a problem while developing my website. As a beginner, I used position: absolute to position almost everything. However, I soon realized that this breaks the natural document flow and makes the layout difficult to manage. Now, I can't properly position elements without disrupting the rest of the content. I need to find a more flexible solution, like using CSS Grid or Flexbox, to create responsive layouts without relying on absolute positioning for everything.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by dalvarez2596** [comment]
Hi @coder-mike-pdf,
It's really cool that you are starting in the world of programming. 
That happens a lot, so my advice is to learn and focus on Flexbox and use it, instead of just using `position: absolute`. There are really good resources like [Flexbox Adventure](https://codingfantasy.com/games/flexboxadventure) or [Flexbox Froggy](https://flexboxfroggy.com/#es) that can help you to learn more deeply about these properties while playing a really entertaining game.

**Reply by ghost** [reply]
Thank you so much for your suggestion! I appreciate it a lot. However, here’s my challenge: the task I set for myself requires very specific positioning, and using Flexbox would alter the layout significantly, which isn't ideal for this project. I’m trying to ensure each element stays precisely where I want it, and position: absolute helps me achieve that.

I understand the benefits of Flexbox, and I’ll definitely check out Flexbox Adventure and Flexbox Froggy for future projects that might need a more flexible layout. Thanks again for the advice!

**Comment by ruhan-ahmad** [comment]
You can try adding position:relative to body tag in HTML, this way all the position absolute will consider body as of its parent & might fix issue for time being or if you have divs that contains child element you can make those div relative that way each child element will connect to their parent div instead of viewport.

Bit if you can update it with flex that way you have more of control on elements.

**Reply by ghost** [reply]
I see what you're saying, and I appreciate the suggestion! The issue, though, is that I don’t want to change the existing content positioning. Switching to position: relative or using Flexbox would shift everything from the specific layout I’ve created. My goal is to maintain absolute control over each element’s exact position without altering the original setup, so I’m sticking with position: absolute for this project. Thanks for the input, though—I’ll keep it in mind for other layouts where flexibility is key!
